### PR TITLE
Add a section about working with Random in native

### LINKED ISSUE #26835: Native compilation: Detected an instance of Random/SplittableRandom class in the image heap - java.security.SecureRandom
### Describe the bug

When i try to Native compile my Application i got this error.

```
Fatal error: org.graalvm.compiler.debug.GraalError: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Detected an instance of Random/SplittableRandom class in the image heap. Instances created during image generation have cached seed values and don't behave as expected.  To see how this object got instantiated use --trace-object-instantiation=java.security.SecureRandom. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
        at com.oracle.graal.pointsto.util.AnalysisFuture.setException(AnalysisFuture.java:49)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:269)
        at com.oracle.graal.pointsto.util.AnalysisFuture.ensureDone(AnalysisFuture.java:63)
        at com.oracle.graal.pointsto.heap.ImageHeapScanner.lambda$postTask$9(ImageHeapScanner.java:611)
        at com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:193)
        at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:177)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Detected an instance of Random/SplittableRandom class in the image heap. Instances created during image generation have cached seed values and don't behave as expected.  To see how this object got instantiated use --trace-object-instantiation=java.security.SecureRandom. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
        at com.oracle.svm.hosted.image.DisallowedImageHeapObjectFeature.error(DisallowedImageHeapObjectFeature.java:173)
        at com.oracle.svm.core.image.DisallowedImageHeapObjects.check(DisallowedImageHeapObjects.java:62)
        at com.oracle.svm.hosted.image.DisallowedImageHeapObjectFeature.replacer(DisallowedImageHeapObjectFeature.java:149)
        at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:582)
        at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:257)
        at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:228)
        at com.oracle.svm.hosted.heap.SVMImageHeapScanner.transformFieldValue(SVMImageHeapScanner.java:126)
        at com.oracle.graal.pointsto.heap.ImageHeapScanner.onFieldValueReachable(ImageHeapScanner.java:331)
        at com.oracle.graal.pointsto.heap.ImageHeapScanner.onFieldValueReachable(ImageHeapScanner.java:310)
        at com.oracle.graal.pointsto.heap.ImageHeapScanner.lambda$computeTypeData$1(ImageHeapScanner.java:153)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        ... 10 more
```

After run 
`mvnw package -Pnative -Dquarkus.native.additional-build-args="--trace-object-instantiation=java.security.SecureRandom"`
 I get the following StackTracke
```
Fatal error: org.graalvm.compiler.debug.GraalError: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Detected an instance of Random/SplittableRandom class in the image heap. Instances created during image generation have cached seed values and don't behave as expected.  Object has been initialized by the com.sap.conn.rfc.engine.GUID class initializer with a trace:
        at java.security.SecureRandom.<init>(SecureRandom.java:224)
        at com.sap.conn.rfc.engine.GUID.<clinit>(GUID.java:38)
. Try avoiding to initialize the class that caused initialization of the object. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
        at com.oracle.graal.pointsto.util.AnalysisFuture.setException(AnalysisFuture.java:49)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:269)
        at com.oracle.graal.pointsto.util.AnalysisFuture.ensureDone(AnalysisFuture.java:63)
        at com.oracle.graal.pointsto.heap.ImageHeapScanner.lambda$postTask$9(ImageHeapScanner.java:611)
        at com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:193)
        at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:177)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Detected an instance of Random/SplittableRandom class in the image heap. Instances created during image generation have cached seed values and don't behave as expected.  Object has been initialized by the com.sap.conn.rfc.engine.GUID class initializer with a trace:
        at java.security.SecureRandom.<init>(SecureRandom.java:224)
        at com.sap.conn.rfc.engine.GUID.<clinit>(GUID.java:38)
. Try avoiding to initialize the class that caused initialization of the object. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
        at com.oracle.svm.hosted.image.DisallowedImageHeapObjectFeature.error(DisallowedImageHeapObjectFeature.java:173)
        at com.oracle.svm.core.image.DisallowedImageHeapObjects.check(DisallowedImageHeapObjects.java:62)
        at com.oracle.svm.hosted.image.DisallowedImageHeapObjectFeature.replacer(DisallowedImageHeapObjectFeature.java:149)
        at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:582)
        at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:257)
        at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:228)
        at com.oracle.svm.hosted.heap.SVMImageHeapScanner.transformFieldValue(SVMImageHeapScanner.java:126)
        at com.oracle.graal.pointsto.heap.ImageHeapScanner.onFieldValueReachable(ImageHeapScanner.java:331)
        at com.oracle.graal.pointsto.heap.ImageHeapScanner.onFieldValueReachable(ImageHeapScanner.java:310)
        at com.oracle.graal.pointsto.heap.ImageHeapScanner.lambda$computeTypeData$1(ImageHeapScanner.java:153)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        ... 10 more
```

When i build with the following additional-build-args in my application.properties
`"--initialize-at-run-time=com.sap.conn.rfc.engine.GUID"`
I got this error
```
Error: Unsupported features in 2 methods
Detailed message:
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: com.sap.jdsr.writer.DsrIPassport. This error is reported at image build time because class com.sap.conn.jco.rt.ClientConnection is registered for linking at image build time by command line
Trace:
        at parsing com.sap.conn.jco.rt.ClientConnection.beginCall(ClientConnection.java:263)
Call path from entry point to com.sap.conn.jco.rt.ClientConnection.beginCall():
        at com.sap.conn.jco.rt.ClientConnection.beginCall(ClientConnection.java:248)
        at com.sap.conn.jco.rt.ClientConnection.execute(ClientConnection.java:1043)
        at com.sap.conn.jco.rt.ClientConnection.execute(ClientConnection.java:940)
        at com.sap.conn.jco.rt.RfcDestination.execute(RfcDestination.java:1140)
        at com.sap.conn.jco.rt.RfcDestination.execute(RfcDestination.java:1114)
        at com.sap.conn.jco.rt.AbapFunction.execute(AbapFunction.java:302)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:207)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:96)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.licenseToken(ConsoleProcessor.java:950)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.update(ConsoleProcessor.java:625)
        at com.zapliance.continuous.service.CownloadService.lambda$createAndStartCownloadTask$0(CownloadService.java:124)
        at com.zapliance.continuous.service.CownloadService$$Lambda$1400/0x00000007c3804b58.run(Unknown Source)
        at java.lang.Thread.run(Thread.java:833)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
        at com.oracle.svm.core.windows.WindowsPlatformThreads.osThreadStartRoutine(WindowsPlatformThreads.java:143)
        at com.oracle.svm.core.code.IsolateEnterStub.WindowsPlatformThreads_osThreadStartRoutine_77cb625316abfaaa1a6261614dc5eb03b33d1f85(generated:0)
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: com.sap.util.monitor.jarm.TaskMonitor. This error is reported at image build time because class com.sap.conn.jco.util.Jarm is registered for linking at image build time by command line
Trace:
        at parsing com.sap.conn.jco.util.Jarm.getRequestMonitor(Jarm.java:69)
Call path from entry point to com.sap.conn.jco.util.Jarm.getRequestMonitor(String, String):
        at com.sap.conn.jco.util.Jarm.getRequestMonitor(Jarm.java:67)
        at com.sap.conn.jco.rt.AbapRepository.queryFunctionTemplate(AbapRepository.java:921)
        at com.sap.conn.jco.rt.AbapRepository.queryFunctionTemplate(AbapRepository.java:549)
        at com.sap.conn.jco.rt.AbapRepository.getFunctionTemplate(AbapRepository.java:775)
        at com.sap.conn.jco.rt.BasicRepository.getFunction(BasicRepository.java:152)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:162)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:96)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.licenseToken(ConsoleProcessor.java:950)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.update(ConsoleProcessor.java:625)
        at com.zapliance.continuous.service.CownloadService.lambda$createAndStartCownloadTask$0(CownloadService.java:124)
        at com.zapliance.continuous.service.CownloadService$$Lambda$1400/0x00000007c3804b58.run(Unknown Source)
        at java.lang.Thread.run(Thread.java:833)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
        at com.oracle.svm.core.windows.WindowsPlatformThreads.osThreadStartRoutine(WindowsPlatformThreads.java:143)
        at com.oracle.svm.core.code.IsolateEnterStub.WindowsPlatformThreads_osThreadStartRoutine_77cb625316abfaaa1a6261614dc5eb03b33d1f85(generated:0)

com.oracle.svm.core.util.UserError$UserException: Unsupported features in 2 methods
Detailed message:
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: com.sap.jdsr.writer.DsrIPassport. This error is reported at image build time because class com.sap.conn.jco.rt.ClientConnection is registered for linking at image build time by command line
Trace:
        at parsing com.sap.conn.jco.rt.ClientConnection.beginCall(ClientConnection.java:263)
Call path from entry point to com.sap.conn.jco.rt.ClientConnection.beginCall():
        at com.sap.conn.jco.rt.ClientConnection.beginCall(ClientConnection.java:248)
        at com.sap.conn.jco.rt.ClientConnection.execute(ClientConnection.java:1043)
        at com.sap.conn.jco.rt.ClientConnection.execute(ClientConnection.java:940)
        at com.sap.conn.jco.rt.RfcDestination.execute(RfcDestination.java:1140)
        at com.sap.conn.jco.rt.RfcDestination.execute(RfcDestination.java:1114)
        at com.sap.conn.jco.rt.AbapFunction.execute(AbapFunction.java:302)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:207)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:96)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.licenseToken(ConsoleProcessor.java:950)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.update(ConsoleProcessor.java:625)
        at com.zapliance.continuous.service.CownloadService.lambda$createAndStartCownloadTask$0(CownloadService.java:124)
        at com.zapliance.continuous.service.CownloadService$$Lambda$1400/0x00000007c3804b58.run(Unknown Source)
        at java.lang.Thread.run(Thread.java:833)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
        at com.oracle.svm.core.windows.WindowsPlatformThreads.osThreadStartRoutine(WindowsPlatformThreads.java:143)
        at com.oracle.svm.core.code.IsolateEnterStub.WindowsPlatformThreads_osThreadStartRoutine_77cb625316abfaaa1a6261614dc5eb03b33d1f85(generated:0)
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: com.sap.util.monitor.jarm.TaskMonitor. This error is reported at image build time because class com.sap.conn.jco.util.Jarm is registered for linking at image build time by command line
Trace:
        at parsing com.sap.conn.jco.util.Jarm.getRequestMonitor(Jarm.java:69)
Call path from entry point to com.sap.conn.jco.util.Jarm.getRequestMonitor(String, String):
        at com.sap.conn.jco.util.Jarm.getRequestMonitor(Jarm.java:67)
        at com.sap.conn.jco.rt.AbapRepository.queryFunctionTemplate(AbapRepository.java:921)
        at com.sap.conn.jco.rt.AbapRepository.queryFunctionTemplate(AbapRepository.java:549)
        at com.sap.conn.jco.rt.AbapRepository.getFunctionTemplate(AbapRepository.java:775)
        at com.sap.conn.jco.rt.BasicRepository.getFunction(BasicRepository.java:152)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:162)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:96)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.licenseToken(ConsoleProcessor.java:950)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.update(ConsoleProcessor.java:625)
        at com.zapliance.continuous.service.CownloadService.lambda$createAndStartCownloadTask$0(CownloadService.java:124)
        at com.zapliance.continuous.service.CownloadService$$Lambda$1400/0x00000007c3804b58.run(Unknown Source)
        at java.lang.Thread.run(Thread.java:833)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
        at com.oracle.svm.core.windows.WindowsPlatformThreads.osThreadStartRoutine(WindowsPlatformThreads.java:143)
        at com.oracle.svm.core.code.IsolateEnterStub.WindowsPlatformThreads_osThreadStartRoutine_77cb625316abfaaa1a6261614dc5eb03b33d1f85(generated:0)

        at com.oracle.svm.core.util.UserError.abort(UserError.java:84)
        at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:249)
        at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:754)
        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:558)
        at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:515)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:407)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:585)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:615)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Unsupported features in 2 methods
Detailed message:
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: com.sap.jdsr.writer.DsrIPassport. This error is reported at image build time because class com.sap.conn.jco.rt.ClientConnection is registered for linking at image build time by command line
Trace:
        at parsing com.sap.conn.jco.rt.ClientConnection.beginCall(ClientConnection.java:263)
Call path from entry point to com.sap.conn.jco.rt.ClientConnection.beginCall():
        at com.sap.conn.jco.rt.ClientConnection.beginCall(ClientConnection.java:248)
        at com.sap.conn.jco.rt.ClientConnection.execute(ClientConnection.java:1043)
        at com.sap.conn.jco.rt.ClientConnection.execute(ClientConnection.java:940)
        at com.sap.conn.jco.rt.RfcDestination.execute(RfcDestination.java:1140)
        at com.sap.conn.jco.rt.RfcDestination.execute(RfcDestination.java:1114)
        at com.sap.conn.jco.rt.AbapFunction.execute(AbapFunction.java:302)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:207)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:96)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.licenseToken(ConsoleProcessor.java:950)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.update(ConsoleProcessor.java:625)
        at com.zapliance.continuous.service.CownloadService.lambda$createAndStartCownloadTask$0(CownloadService.java:124)
        at com.zapliance.continuous.service.CownloadService$$Lambda$1400/0x00000007c3804b58.run(Unknown Source)
        at java.lang.Thread.run(Thread.java:833)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
        at com.oracle.svm.core.windows.WindowsPlatformThreads.osThreadStartRoutine(WindowsPlatformThreads.java:143)
        at com.oracle.svm.core.code.IsolateEnterStub.WindowsPlatformThreads_osThreadStartRoutine_77cb625316abfaaa1a6261614dc5eb03b33d1f85(generated:0)
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: com.sap.util.monitor.jarm.TaskMonitor. This error is reported at image build time because class com.sap.conn.jco.util.Jarm is registered for linking at image build time by command line
Trace:
        at parsing com.sap.conn.jco.util.Jarm.getRequestMonitor(Jarm.java:69)
Call path from entry point to com.sap.conn.jco.util.Jarm.getRequestMonitor(String, String):
        at com.sap.conn.jco.util.Jarm.getRequestMonitor(Jarm.java:67)
        at com.sap.conn.jco.rt.AbapRepository.queryFunctionTemplate(AbapRepository.java:921)
        at com.sap.conn.jco.rt.AbapRepository.queryFunctionTemplate(AbapRepository.java:549)
        at com.sap.conn.jco.rt.AbapRepository.getFunctionTemplate(AbapRepository.java:775)
        at com.sap.conn.jco.rt.BasicRepository.getFunction(BasicRepository.java:152)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:162)
        at com.zapliance.modules.duniverseCownload.SAPTableQuerySimple.queryTable(SAPTableQuerySimple.java:96)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.licenseToken(ConsoleProcessor.java:950)
        at com.zapliance.modules.duniverseCownload.ConsoleProcessor.update(ConsoleProcessor.java:625)
        at com.zapliance.continuous.service.CownloadService.lambda$createAndStartCownloadTask$0(CownloadService.java:124)
        at com.zapliance.continuous.service.CownloadService$$Lambda$1400/0x00000007c3804b58.run(Unknown Source)
        at java.lang.Thread.run(Thread.java:833)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
        at com.oracle.svm.core.windows.WindowsPlatformThreads.osThreadStartRoutine(WindowsPlatformThreads.java:143)
        at com.oracle.svm.core.code.IsolateEnterStub.WindowsPlatformThreads_osThreadStartRoutine_77cb625316abfaaa1a6261614dc5eb03b33d1f85(generated:0)

        at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:129)
        at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:751)
        ... 6 more
```

### Expected behavior

After adding the --initialize-at-run-time=com.sap.conn.rfc.engine.GUID it should work i guess

### Actual behavior

_No response_

### How to Reproduce?

I can give access to the private repository if really needed to selected ppl.

### Output of `uname -a` or `ver`

none

### Output of `java -version`

openjdk version "17.0.3" 2022-04-19 OpenJDK Runtime Environment GraalVM CE 22.1.0 (build 17.0.3+7-jvmci-22.1-b06) OpenJDK 64-Bit Server VM GraalVM CE 22.1.0 (build 17.0.3+7-jvmci-22.1-b06, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.10.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #26835

### CODE REVIEW

File: docs/src/main/asciidoc/writing-native-applications-tips.adoc [hunk 1]
```diff
@@ -330,6 +330,17 @@ and in the case of using the Maven configuration instead of `application.propert
 ----
 ====
 
+=== Working with Random and SplittableRandom
+
+If your native application is initialized at build time and has static `java.util.Random` or `java.util.SplittableRandom` instances then its native build will fail because embedding such instances in native image will result in the same cached seed value be reused which will negatively affect the entropy (randomness) quality of the produced values.
```
Comment (gsmet) [inline docs/src/main/asciidoc/writing-native-applications-tips.adoc:335]: I think this is highly confusing: `If your native application is initialized at build time`.

You don't have the option in Quarkus to initialize "your application" at build time or runtime.

File: docs/src/main/asciidoc/writing-native-applications-tips.adoc [hunk 2]
```diff
@@ -330,6 +330,17 @@ and in the case of using the Maven configuration instead of `application.propert
...
+
+There is a number of techniques that can be used to have static `Random` and `SplittableRandom` instances securely used in native images:
+ * Do the runtime initialization instead of the build-time initialization
+ * Keep the build-time initialization but request a runtime re-inititalization of the affected classes
+ * Substitute instances of `Random` and `SplittableRandom` with `null`
```
Comment (gsmet) [inline docs/src/main/asciidoc/writing-native-applications-tips.adoc:340]: How is it a workable option? The chances that they are not actually used are extremely narrow.

**Review by gsmet — CHANGES_REQUESTED** [review]
Hmmm, I agree we should have something but I don't like the wording of this proposal that much. I'll try to come up with something.

File: docs/src/main/asciidoc/writing-native-applications-tips.adoc (hunk 2 above)
Comment (sberyozkin) [reply docs/src/main/asciidoc/writing-native-applications-tips.adoc:340]: @gsmet For example, there were `Random` instances used from a test algorithm code in BC FIPS case, https://github.com/quarkusio/quarkus/blob/main/extensions/security/runtime/src/main/java/io/quarkus/security/runtime/graal/BouncyCastleSubstitutions.java#L24, cases like that...

File: docs/src/main/asciidoc/writing-native-applications-tips.adoc (hunk 1 above)
Comment (sberyozkin) [reply docs/src/main/asciidoc/writing-native-applications-tips.adoc:335]: @gsmet Sure, this is a general point I've tried to convey from the tutorial.

So. may be just `If your native application has static ...`, omitting `is initialized at build time`, please suggest an update.

**Comment by zakkak** [comment]
> Hmmm, I agree we should have something but I don't like the wording of this proposal that much. I'll try to come up with something.

I agree. I will wait for a new version or @gsmet's proposal to review. Thanks for the PR @sberyozkin.

**Comment by sberyozkin** [comment]
Np, lets tune it to make it technically precise

**Comment by gsmet** [comment]
Superseded by https://github.com/quarkusio/quarkus/pull/27106 .

**Pull request closed without merging**